irohad/ordering/impl/on_demand_ordering_gate.hpp

```diff
@@ -32,34 +41,13 @@ namespace iroha {
      */
     class OnDemandOrderingGate : public network::OrderingGate {
      public:
-      /**
-       * Represents storage modification. Proposal round increment
-       */
-      struct BlockEvent {
-        /// next round number
-        consensus::Round round;
-        /// hashes of processed transactions
-        cache::OrderingGateCache::HashesSetType hashes;
-
-        std::string toString() const;
-      };
-
-      /**
-       * Represents no storage modification. Reject round increment
-       */
-      struct EmptyEvent {
-        /// next round number
-        consensus::Round round;
-
-        std::string toString() const;
-      };
-
-      using BlockRoundEventType = boost::variant<BlockEvent, EmptyEvent>;
-
       OnDemandOrderingGate(
           std::shared_ptr<OnDemandOrderingService> ordering_service,
           std::shared_ptr<transport::OdOsNotification> network_client,
-          rxcpp::observable<BlockRoundEventType> events,
+          rxcpp::observable<std::shared_ptr<shared_model::interface::Block>>
```

Comment:
It seems that the implementation requires only hashes of transactions. I guess better to restrict the interface and pass only observable with the list of committed\rejected hashes.

Reply:
moved tx hashes extraction to ordering_init